src/api/covariates/vibrant_oceans.py

```diff
@@ -0,0 +1,157 @@
+    NUM_PORTS = "num_ports"
+    REEF_VALUE = "reef_value"
+    CUMUL_SCORE = "cumul_score"
+
+    COLUMNS = (
+        SCORE,
+        SCORE_CN,
+        SCORE_CY,
+        SCORE_PFC,
+        SCORE_TH,
+        SCORE_TR,
+        GRAV_NC,
+        SEDIMENT,
+        NUTRIENT,
+        POP_COUNT,
+        NUM_PORTS,
+        REEF_VALUE,
+        CUMUL_SCORE,
+    )
+
+    @property
+    def display_name_lookup(self):
+        return {
+            f"vot_{self.SCORE}": "Vibrant Oceans Climate: Composite Score",
+            f"vot_{self.SCORE_CN}": "Vibrant Oceans Climate: Connectivity",
+            f"vot_{self.SCORE_CY}": "Vibrant Oceans Climate: Cyclone Risk",
+            f"vot_{self.SCORE_PFC}": "Vibrant Oceans Climate: Thermal Future",
+            f"vot_{self.SCORE_TH}": "Vibrant Oceans Climate: Thermal History",
+            f"vot_{self.SCORE_TR}": "Vibrant Oceans Climate: Recent Stress",
+            f"vot_{self.GRAV_NC}": "Vibrant Oceans Fishing: Market Pressure",
+            f"vot_{self.SEDIMENT}": "Vibrant Oceans Pollution: Sedimentation",
+            f"vot_{self.NUTRIENT}": "Vibrant Oceans Pollution: Nutrients",
+            f"vot_{self.POP_COUNT}": "Vibrant Oceans Coastal Development: Human Population",
+            f"vot_{self.NUM_PORTS}": "Vibrant Oceans Industrial Development: Ports",
+            f"vot_{self.REEF_VALUE}": "Vibrant Oceans Tourism: Reef Value",
+            f"vot_{self.CUMUL_SCORE}": "Vibrant Oceans Climate",
+        }
+
+    def _table_exists(self):
+        sql = """
+            SELECT EXISTS (
+                SELECT 1
+                FROM
+                    information_schema.tables
+                WHERE
+                    table_schema = 'covariates' AND
+                    table_name = 'allreef'
+            )
+        """
+        with connection.cursor() as cursor:
+            cursor.execute(sql)
+            record = cursor.fetchone()
+            return record[0]
+
+    def _fetch(self, x: float, y: float, radius: float, request_datetime: datetime):
+
+        if self._table_exists() is not True:
+            return None
+
+        _covariate_cols = ", ".join(self.COLUMNS)
+        _sum_covariate_cols = ", ".join(
+            f"SUM({c} * partial) as {c}" for c in self.COLUMNS
+        )
+        sql_template = f"""
+            WITH allreef_clipped AS (
+                SELECT
+                    {_covariate_cols},
+                    ST_Area(
+                        ST_Intersection(
+                            ST_Buffer(
+                                ST_GeomFromText('SRID=4326;POINT(%(x)s %(y)s)')::geography,
+                                %(radius)s
+                            ),
+                            allreef.geom::geography
+                        ),
+                        true
+                    ) /
+                    ST_Area(
+                        ST_Buffer(
+                            ST_GeomFromText('SRID=4326;POINT(%(x)s %(y)s)')::geography,
```

Comment:
Maybe format this and the above instances the same as below, on one line

Reply:
I think the line length gets pretty long then

Reply:
OK no big deal -- but I had to read these a couple times, whereas 109 I got in one glance :-)